File: app/src/main/java/com/jack/learn/aidl/Person.java
package com.jack.learn.aidl;

import android.os.Parcel;
import android.os.Parcelable;

public class Person implements Parcelable{


    public String name;
    public int age;

    public Person() {

    }

    Person(String name, int age) {
        this.name = name;
        this.age = age;
    }

    protected Person(Parcel in) {
        name = in.readString();
        age = in.readInt();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(name);
        dest.writeInt(age);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<Person> CREATOR = new Creator<Person>() {
        @Override
        public Person createFromParcel(Parcel in) {
            return new Person(in);
        }

        @Override
        public Person[] newArray(int size) {
            return new Person[size];
        }
    };
}

File: app/src/main/java/com/jack/learn/kotlin/Person.kt
package com.jack.learn.kotlin

import android.util.Log

/**
 * 主构造函数执行优先级最高、init代码块次之、次构造函数执行优先级最低
 */
// 主构造函数
class Person(name: String) {
    //主构造函数的一部分
    var nameT: String = name

    init {
        Log.d("wangjie","init代码块 name: $nameT")
    }

    //次构造函数
    constructor(name: String, age: Int):this(name) {
        Log.d("wangjie","name: $nameT")
        Log.d("wangjie","secondary constructor")
    }


}